src/private.rs

```diff
@@ -110,6 +111,43 @@ impl SecretKey {
         );
         proof
     }
+
+    /// Prove correct decryption without depending on the zkp toolkit, which
+    /// uses Merlin for Transcripts. The latter is hard to mimic in solidity
+    /// smart contracts. To this end, we define this alternative proof of correct
+    /// decryption which allows us to proceed with the verification in solidity.
+    /// This function should only be used in the latter case. If the verification is
+    /// performed in rust, `prove_correct_decryption` function should be used.
+    pub fn prove_correct_decryption_no_Merlin(
+        &self,
+        ciphertext: &Ciphertext,
+        message: &RistrettoPoint,
+    ) -> ((CompressedRistretto, CompressedRistretto), Scalar) {
+        let pk = PublicKey::from(self);
+        let announcement_random = Scalar::random(&mut OsRng);
+        let announcement_base_G = announcement_random * RISTRETTO_BASEPOINT_POINT;
+        let announcement_base_ctxtp0 = announcement_random * ciphertext.points.0;
+
+        let challenge = Scalar::from_hash(
```

Comment:
Can you move this part into a separate function so that public.rs doesn't need to duplicate it?

Reply:
Yep, makes sense.